clang/test/CodeGenSYCL/device_has.cpp

```diff
@@ -1,38 +1,32 @@
 // RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -triple spir64-unknown-unknown -disable-llvm-passes -emit-llvm %s -o - | FileCheck %s
 
-// Tests for IR of device_has(aspect, ...) attribute and
-// !sycl_used_aspects metadata
+// Tests for IR of device_has(aspect, ...) attribute
```

Comment:
With the modifications in this test we no longer has a FE test checking aspect propagation. I would suggest removing `-disable-llvm-passes` and keeping the test as-is

Reply:
Also this is what concerns me. Are we not changing behavior of compile time properties with this change? Shouldn't functionality remain the same irrespective of optimization level?

Reply:
`-disable-llvm-passes` is internal option to completely disable passes regardless of optimization level. It should only be used by developers during debugging or some experiments.

> With the modifications in this test we no longer has a FE test checking aspect propagation.

I think that is fine, because FE codegen does not propagate aspects. That is done through the pass and tested separately as LIT for that pass. Integration test which checks combination of both FE codegen and the pass is present in `sycl/test/check_device_code/device_has.cpp`